userspace/openpilot_python_dependencies.sh

```diff
@@ -2,20 +2,19 @@
 
 echo "Installing python for openpilot"
 
-# Install pyenv
-export PYENV_ROOT="/usr/local/pyenv"
-curl https://pyenv.run | bash
-export PATH="$PYENV_ROOT/bin:$PATH"
-eval "$(pyenv init -)"
+if ! command -v "uv" > /dev/null 2>&1; then
+  echo "installing uv..."
+  curl -LsSf https://astral.sh/uv/install.sh | sh
+  UV_BIN='$HOME/.cargo/env'
+  ADD_PATH_CMD=". \"$UV_BIN\""
+  eval $ADD_PATH_CMD
+fi
 
 PYTHON_VERSION="3.11.4"
 if [ "$(uname -p)" == "aarch64" ]; then
-  pyenv install --verbose $PYTHON_VERSION
+  uv python install 3.11.4
 else
-  MAKEFLAGS="-j1" MAKE_OPTS="-j1" taskset --cpu-list 0 pyenv install --verbose $PYTHON_VERSION
+  MAKEFLAGS="-j1" MAKE_OPTS="-j1" taskset --cpu-list 0 uv python install --verbose $PYTHON_VERSION
 fi
-
-echo "Setting global python version"
-pyenv global $PYTHON_VERSION
-
-pip3 install --no-cache-dir --upgrade pip uv
+uv venv
```

Comment:
the dependencies will need to be installed globally

Reply:
From what I have discovered, uv can only do managed installations of python. According to https://github.com/astral-sh/uv/issues/1526#issuecomment-1969930462, --python flag can be used to create virtualenv to arbitrary Python interpreter which can be system path. Creating virtual env pointing to system python can achieve installing dependencies system-wide.